### DISCUSSION TITLE
evince: cannot print: /bin/sh not found

### DISCUSSION THREAD

**Discussion by githlp** [netblue30/firejail#4969] [Q&A] [answered]
Hi - me again.

Could not figure out how to get/persuade evince to send a document to printer.

The error message reads
`child process >>/bin/sh<< cannot be run. File or dir not found`

Whitelisting/NoBlacklisting of /bin/sh in evince.local did not help..

**Comment by rusty-snake** [comment, marked as answer]
```
include allow-bin-sh.inc
private-bin sh
```

**Comment by netblue30** [comment]
I put a fix in here: https://github.com/netblue30/firejail/commit/f347e88cc7c3d6d6dddce51a344adf8200c00773

Thanks for the bug, give it a try.

**Comment by netblue30** [comment]
new fix here: https://github.com/netblue30/firejail/commit/30e5c12407ffdb47a4780901df9bd97a306ddbe0

**Comment by githlp** [comment]
Thanks, documents are sent to the printer again.

I still have these warning messages though:
```
evince:100): dbind-WARNING **: 17:50:56.752: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-UbcgDkREAW: file or dir not found
Error registering document: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown

** (evince:100): WARNING **: 17:51:05.302: Couldn't connect to D-Bus system bus, connection lost: no permission

** (evince:100): WARNING **: 17:51:05.305: failed to contact colord: connection lost: no permission
```

**Reply by rusty-snake** [reply]
https://github.com/netblue30/firejail/discussions/4737

**Reply by githlp** [reply]
Now all of a sudden evince will not open attached pdf from evolution due to some permissions problem.
It worked before. I did not change the evince.local conf and no new firejail versions/profiles were released inbetween.
Error reads:
`error opening file »/home/ronald/.cache/evolution/tmp/evolution-ronald-sX8WUU/Rechnung VKRA22A0120449.pdf«: no permission`
The whole path as well as the file itself is readable for my user and my group though.
Any hints?

**Reply by rusty-snake** [reply]
```
noblacklist ${HOME}/.cache/evolution
ignore private-cache
```

**Comment by githlp** [comment]
oh, my bad:
`noblacklist ${HOME}/` 
will not work, but
`noblacklist ${HOME}/.cache/evolution`
works.
